### DISCUSSION TITLE
NODE_ENV

### DISCUSSION THREAD

**Discussion by milesje** [mobxjs/mobx#2714] [Q&A] [answered]
So I just updated to Mobx v6.X and had issues with mobx complaining about the NODE_ENV not being set.... so after some searching the only option I found that actually works was to hard code this `globalThis.process = { env: { NODE_ENV: 'production' } };` in my index.html file. I can provide a test application via zip file or github repo if needed. But, it does seem a little unusual for have to set this value for a Web Application. Is there another way to get this value set that I just don't know about that would be better than hard coding it into the index.html file?

Here is a test web application to see what I'm talking about.
https://github.com/milesje/navigo-test

**Comment by mweststrate** [comment]
Depends on whether you are using MobX using a script tag or a bundler.
Bundlers should take care of setting it, if you use a script tag, make sure
to pick the UMD build.

**Reply by urugator** [reply]
Afaik rollup does nearly nothing without plugins except merging the sources.
@milesje https://github.com/mobxjs/mobx/issues/2564#issuecomment-759415865 could help

**Comment by milesje** [comment]
It is being imported like `import { observable, action } from 'mobx'` in a .ts file, but I'm not using a bundler (during development). I'm just using the TypeScript compiler to compile everything to JavaScript and then use @web/dev-server to run the output of tsc in watch mode.

Here is the 'start' script in my package.json

`"start": "tsc && concurrently -k -r \"tsc --watch --preserveWatchOutput\" \"wds\"",`

**Reply by danielkcz** [reply]
And where are you running that compiled code? In NodeJS?

**Reply by milesje** [reply]
In a browser

**Reply by milesje** [reply]
@FredyC And even when using rollup to bundle everything together I still get the error `Uncaught ReferenceError: process is not defined`
![image](https://user-images.githubusercontent.com/537359/104644760-d2a21f00-5673-11eb-9ed5-177fbd6912e0.png)
And this is running the bundled version that was output from rollup.

**Reply by milesje** [reply]
OK, I found a workaround (maybe a solution).... I can add a rollup plugin to @web/dev-server to inject the process.env.NODE_ENV, but this really seems like a hacky way of doing things. Also when bundling with rollup I also have to add this same plugin to get it to inject the process.env.NODE_ENV.

**Comment by milesje** [comment, marked as answer]
OK, I found a workaround (maybe a solution).... I can add a rollup plugin to @web/dev-server to inject the process.env.NODE_ENV, but this really seems like a hacky way of doing things. Also when bundling with rollup I also have to add this same plugin to get it to inject the process.env.NODE_ENV.

**Comment by mweststrate** [comment]
Rollup doesn't default to bundling modules for the web specifically. To
bundle for browser consumption, you will need to use either the prebuild
umd version of mobx, or use a webbundler like webpack or parcel, or indeed
further configure rollup with plugins. This is further working as intended,
as without this setup, bundlers wouldn't be able to pick separate versions
for prod or debug.

Op do 14 jan. 2021 21:38 schreef Jacob Miles <notifications@github.com>:
